tests/selenium/dnt_test.py

```diff
@@ -212,28 +225,15 @@ def test_should_not_record_nontracking_domains(self):
 
     def test_first_party_dnt_header(self):
         TEST_URL = "https://httpbin.org/get"
-
-        self.load_url(TEST_URL)
-
-        headers = json.loads(
-            self.driver.find_element_by_tag_name('body').text
-        )['headers']
-
+        headers = retry_until(partial(self.get_first_party_headers, TEST_URL))
         self.assertIn('Dnt', headers, "DNT header should have been present")
```

Comment:
It doesn't matter a whole lot, but `get_first_party_headers` will return False if it can't parse anything, and this check will throw a TypeError instead of failing.

Reply:
Yep, I noticed that too but I'm not sure we need to do anything. We could improve this by adding another assertion right after getting headers. "Did we succeed in getting headers?" This would replace the low-level error with a user-friendly test failure. I was just felt like it wasn't important. It seems important to differentiate between not having DNT in headers and not having headers at all, and we do that already. What do you think?

Reply:
I think a an assert(type(headers) == list) or similar is easy enough to add that it's worth it, if only to make it clear with code that we want to differentiate between those two failure cases.

Reply:
Err, yes! Forgot Python does empty objects/lists differently from JS.

Reply:
Although I'm not sure when an empty dictionary will be returned here.